lib/rester/service/resource/params.rb

```diff
@@ -22,17 +24,34 @@ def strict?
         !!options[:strict]
       end
 
+      def frozen?
+        !!@_frozen
+      end
+
       def freeze
         @_validators.freeze
         @_required_fields.freeze
         @_defaults.freeze
         @_all_fields.freeze
+        @_frozen = true
```

Comment:
I'm realizing there's a better way to do this since freeze is built in.

I was thinking, what's to stop someone from overwriting the `@_frozen` instance variable?

Freezing is built in, so instead of doing `@_frozen` do:

```ruby
def freeze
  @_validators.freeze
  @_required_fields.freeze
  @_defaults.freeze
  @_all_fields.freeze
  super
end
```

The `frozen?` method is already built in and calling `super` will prevent any of the instance variables from being modified: 
```ruby
obj.instance_variable_set(:@_all_fields, 'something else!')
```

Calling `freeze` on `@_all_fields` prevents that object from being modified, but it doesn't prevent the instance variable itself from being overwritten.